services/motion/builtin/move_attempt.go

```diff
@@ -49,19 +49,23 @@ func newMoveAttempt(ctx context.Context, request *moveRequest) *moveAttempt {
 	var waypointIndex atomic.Int32
 	waypointIndex.Store(1)
 
-	return &moveAttempt{
+	ma := &moveAttempt{
 		ctx:               cancelCtx,
 		cancelFn:          cancelFn,
 		backgroundWorkers: &backgroundWorkers,
 
 		request:      request,
 		responseChan: make(chan moveResponse),
 
-		position: newReplanner(time.Duration(1000/request.config.PositionPollingFreqHz)*time.Millisecond, request.deviatedFromPlan),
-		obstacle: newReplanner(time.Duration(1000/request.config.ObstaclePollingFreqHz)*time.Millisecond, request.obstaclesIntersectPlan),
-
 		waypointIndex: &waypointIndex,
 	}
+	if request.config.PositionPollingFreqHz > 0 {
+		ma.position = newReplanner(time.Duration(1000/request.config.PositionPollingFreqHz)*time.Millisecond, request.deviatedFromPlan)
```

Comment:
I thought we used a default when the value was zero?  If that is the case I don't know if this is doing anything

Reply:
It does things for tests that construct the struct manually.

Also, this will be changing soon anyway with @nicksanford 's incoming PR. I will be updating to use that instead of this.